src/helics/core/CoreBroker.cpp

```diff
@@ -2590,16 +2606,25 @@ std::string CoreBroker::generateQueryAnswer(const std::string& request)
     if (mi != mapIndex.end()) {
         auto index = mi->second.first;
         if (isValidIndex(index, mapBuilders) && !mi->second.second) {
-            if (std::get<0>(mapBuilders[index]).isCompleted()) {
-                return std::get<0>(mapBuilders[index]).generate();
+            auto& builder = std::get<0>(mapBuilders[index]);
+            if (builder.isCompleted()) {
+                auto center = generateMapObjectCounter();
+                if (center == builder.getCounterCode()) {
+                    return std::get<0>(mapBuilders[index]).generate();
```

Comment:
Same comment here as in the other file.
```suggestion
                    return builder.generate();
```